Packs/illuminate/Integrations/Analyst1/Analyst1.py

```diff
@@ -511,9 +1127,64 @@ def analyst1_batch_check_post(client: Client, args: dict) -> dict | None:
         "runpath": runpath,
     }
 
+    # Support both comma and newline delimiters for values parameter
+    # POST API expects newline-separated format (file upload)
+    if values and "," in values and "\n" not in values:
+        # Comma-delimited input - split and convert to newline-delimited
+        values = "\n".join(val.strip() for val in values.split(","))
+
     raw_data = client.post_batch_search(values)
     # assume succesful result or client will have errored
     if len(raw_data["results"]) > 0:
+        # Group results by searchedValue (the actual value searched) to handle multiple entity types per indicator
```

Comment:
The logic for processing batch results appears to be duplicated from the `analyst1_batch_check_command` function. To improve maintainability, consider extracting this shared logic into a dedicated helper function.